crates/bevy_ecs/src/relationship/relationship_query.rs

```diff
@@ -9,24 +11,26 @@ use smallvec::SmallVec;
 
 use super::SourceIter;
 
-impl<'w, 's, D: QueryData, F: QueryFilter> Query<'w, 's, D, F> {
+impl<'w, 's, D: QueryData, F: QueryFilter, S: Deref<Target = QueryState<D, F>>>
+    Query<'w, 's, D, F, S>
```

Comment:
I can see how adding `S` positions us better to support "uncached queries", but I don't have an easy answer to how we get from `S: Deref<Target = QueryState<D, F>>` to a type constraint that could accommodate uncached queries (which certainly won't be able to deref to QueryState).

What is the vision there?